runeq/resources/common.py

```diff
@@ -141,19 +144,42 @@ def get(self, id: str) -> ItemBase:
             f"{self._item_class.__name__} with ID {id}"
         )
 
-    def add(self, *items: ItemBase):
+    def add(self, items: Union[ItemBase, Iterable[ItemBase]]):
         """
-        Add item(s) to this set.
+        Add item(s) to this set. Accepts a single item OR an interable of
+        items that have the same type as the item class (including another
+        ItemSet of this type).
 
         """
+        if type(items) is self._item_class:
+            items = [items]
+
         for item in items:
             if type(item) is not self._item_class:
                 raise TypeError(
-                    f'cannot add {str(item)}; must be type {self._item_class}'
+                    f'cannot add {str(item)}; must be type '
+                    f'{self._item_class.__name__}'
                 )
 
             self._items[item.id] = item
 
+    def __add__(self, other: 'ItemSet'):
```

Comment:
Sets seem to have different ways of adding items, which we might want to stick to if we want to be more pythonic. This might be overkill but other set operations are [summarized here](https://tutorial.eyehunts.com/python/python-set-extend-example-code/), and note that `+` isn't an operation:
```
A = {0, 2, 4, 6}
B = {1, 2, 3, 4}

print("Union :", A | B)

print("Intersection :", A & B)

print("Difference :", A - B)

# elements not present both sets
print("Symmetric difference :", A ^ B)
```

Reply:
Yeah, that's true... but I don't know how much that matters for this, it's already a distinct class. My instinct is that `+` operator would be pretty intuitive for people to use -- perhaps more than set operations. That said, I think supporting set operations would _also_  be interesting!

Either way, I think this is worth more of a discussion! I'm going to remove the `__add__` method. 

One other thing I'll say, though, is I'm also breaking away from the set functionality with the `.add()` method (though it didn't conform before, either). `add` on a Python set accepts exactly one argument and it will add that argument to the set -- it doesn't iterate over anything. There is also an `update` method on Python sets that accepts an iterable. What if we split out the functionality on `ItemSet` into comparable methods? I.e.

```python
def add(self, ItemBase):
    """Add a single item to the collection"""

def update(self, Iterable[ItemBase]):
    """Add each item in an iterable to the collection"""
```